python/osrd_schemas/scripts/truncate_infra.py

```diff
@@ -0,0 +1,376 @@
+from typing import Mapping, Sequence
+
+from shapely import LineString, Polygon
+
+from osrd_schemas.infra import (
+    BufferStop,
+    BufferStopReference,
+    Detector,
+    Direction,
+    Electrification,
+    Endpoint,
+    NeutralSection,
+    OperationalPoint,
+    RailJsonInfra,
+    Route,
+    SpeedSection,
+    Switch,
+    SwitchType,
+    TrackRange,
+)
+from osrd_schemas.switch_type import CROSSING, DOUBLE_SLIP_SWITCH, LINK, POINT_SWITCH
+
+SWITCH_TYPES: Mapping[str, SwitchType] = {
+    "point_switch": POINT_SWITCH,
+    "link": LINK,
+    "crossing": CROSSING,
+    "double_slip_switch": DOUBLE_SLIP_SWITCH,
+}
+
+
+def truncate_infra(infra: RailJsonInfra, bbox: Polygon) -> RailJsonInfra:
```

Comment:
Ditto for all `bbox`:
```suggestion
def truncate_infra(infra: RailJsonInfra, boundary: Polygon) -> RailJsonInfra:
```

Reply:
I’ve updated the commit https://github.com/OpenRailAssociation/osrd/pull/13526/commits/c23fa520ccec5e3316fd069c99640b5aefb2d3d4. I’ve also rebased the PR.